cs/src/Contracts/TunnelAccessControlEntryType.cs

```diff
@@ -57,5 +57,12 @@ public enum TunnelAccessControlEntryType
         /// access to the tunnel.
         /// </summary>
         IPAddressRanges,
+
+        /// <summary>
+        /// The access control entry is a list of service tags representing Azure services that
+        /// are allowed or denied access to the tunnel. These are resolve to IP address ranges by
+        /// the tunnel service.
+        /// </summary>
+        ServiceTag,
```

Comment:
Would it make sense to _not_ add a new entry type, and use `IPAddressRanges` entry type also for service tags? Then the range format depends on whether the provider name is "ipv4", "ipv6", or "service-tag".

Reply:
We can, and I've gone back and forth on it. Having a separate entry just makes the logic a little cleaner when we do the actual access calculation. If one entry seems better to you I'm fine changing it.

Reply:
Whichever way we go here, I'd like to support multiple service tags.
Is that already accounted for?

Reply:
Yes, the name was meant to be plural here. Should be fixed now.

Reply:
I'd lean toward using the `IPAddressRanges` type, since services tags are just another way to specify IP ranges. Unless you think it will really be a problem.

> Whichever way we go here, I'd like to support multiple service tags. Is that already accounted for?

Yes, access control entry objects already support an array of "subjects" per entry. For `Type=IPAddressRanges` and `Provider=service-tag`, each subject array item would be one service tag.

Reply:
Updated to use `IPAddressRanges` for both.